### ISSUE TITLE
IDE0044 "Make field readonly" false positive when used in ref return

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#33009]
**Version Used**: 
16.0.0 Preview 2.0

**Steps to Reproduce**:
```csharp
public class C
{
    private int a, b; // IDE0044
    private ref int select(bool c) => ref (c ? ref a : ref b);
    public void Test() => select(true) = 3;
}
```

**Expected Behavior**:
The members are mutated. Marking them readonly breaks the code, so IDE0044 shouldn't trigger.

**Actual Behavior**:
IDE0044 is falsely triggered.

**Comment by kindermannhubert** [comment]
Same here:
```c#
class C
{
    private int x; // IDE0044

    public void Foo()
    {
        ref var xRef = ref x;
        xRef = 7;
    }
}
```

**Comment by stephentoub** [comment]
Just hit this as well.
```C#
class Program
{
    private int _field;

    public ref int Method() => ref _field;

    static void Main() => new Program().Method() = 42;
}
```

**Comment by mavasani** [comment]
Re-opening based on https://github.com/dotnet/roslyn/pull/45288/files#diff-5ea25649ac3f72b91f34042ee3b8962aR1712-R1724

Tagging @Youssef1313

**Comment by sharwell** [comment]
@mavasani that issue is different. This one only relates to ref _returns_.

**Comment by mavasani** [comment]
@Youssef1313 Can you file a new issue and use that for skipping your test?

**Comment by Youssef1313** [comment]
@mavasani, Sure. I pointed to this issue at first because its link existed as a comment for the disabled warning.

I'll open a new one now.
